From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kai Bolay <kai@bolay.de>
Date: Mon, 3 Aug 2026 19:43:00 -0400
Subject: [PATCH] 9452: Ensure reasonable PNG chunk size

Prevent out of memory error due to invalid input data.
8 MB ought to be enough for an iOS icon.
---
 .../src/main/java/ghidra/file/formats/ios/png/PNGChunk.java    | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Ghidra/Features/FileFormats/src/main/java/ghidra/file/formats/ios/png/PNGChunk.java b/Ghidra/Features/FileFormats/src/main/java/ghidra/file/formats/ios/png/PNGChunk.java
index 8c8fb0dfba..84fb48a278 100644
--- a/Ghidra/Features/FileFormats/src/main/java/ghidra/file/formats/ios/png/PNGChunk.java
+++ b/Ghidra/Features/FileFormats/src/main/java/ghidra/file/formats/ios/png/PNGChunk.java
@@ -40,6 +40,9 @@ public class PNGChunk implements StructConverter {
 	 */
 	public PNGChunk(BinaryReader reader) throws IOException {
 		length = reader.readNextInt();
+		if (length < 0 || length > 8 * 1024 * 1024) {
+			throw new IllegalArgumentException("Unreasonable PNG chunk length: " + length);
+		}
 		chunkID = reader.readNextInt();
 		data = reader.readNextByteArray(length);
 		crc32 = reader.readNextInt();
-- 
2.45.1

